tpm2/tpm2.go

```diff
@@ -418,6 +418,57 @@ func CreatePrimary(rw io.ReadWriter, owner tpmutil.Handle, sel PCRSelection, par
 	return decodeCreatePrimary(resp)
 }
 
+// CreatePrimaryEx initializes the primary key in a given hierarchy.
+// This function differs from CreatePrimary in that all response elements
+// are returned, and they are returned in relatively raw form.
+//
+// The first return value is a transient handle to the loaded key.
+// The second return value is the public part of the generated key.
+// The third return value is the creation data of the key.
+// The fourth return value is the digest of the creation data.
+// The fifith return value is the creation ticket.
+// The sixth return value is the creation name.
+func CreatePrimaryEx(rw io.ReadWriter, owner tpmutil.Handle, sel PCRSelection, parentPassword, ownerPassword string, pub Public) (tpmutil.Handle, []byte, []byte, []byte, Ticket, []byte, error) {
+	cmd, err := encodeCreate(owner, sel, parentPassword, ownerPassword, nil /*inSensitive*/, pub)
+	if err != nil {
+		return 0, nil, nil, nil, Ticket{}, nil, err
+	}
+	resp, err := runCommand(rw, TagSessions, cmdCreatePrimary, tpmutil.RawBytes(cmd))
+	if err != nil {
+		return 0, nil, nil, nil, Ticket{}, nil, err
+	}
+
+	var out struct {
```

Comment:
Move all of this decoding into `decodeCreatePrimary` or `decodeCreate`
Also, you can probably use a bunch of vars instead of a struct